sql/repl_failsafe.cc

```diff
@@ -183,7 +183,7 @@ static my_bool show_slave_hosts_callback(THD *thd, Protocol *protocol)
     protocol->prepare_for_resend();
     protocol->store(si->server_id);
     protocol->store(si->host, strlen(si->host), &my_charset_bin);
-    if (opt_show_slave_auth_info)
+    if (thd->variables.show_slave_auth_info)
```

Comment:
Is it possible for `@@SESSION.show_slave_auth_info` to change in the middle of a `SHOW REPLICA HOSTS`?

Reply:
Adding a system varaible does (or at least *should*) not affect how it’s set from an options.
My variable tests uses `.opt` files to assert that the variables default to what’s set from the options.

Reply:
For the perspective of “scope”, though, I vaguely recall finding another variale or two (besides `@@replicate_same_server_id`, if that’s one of them) that make sense to be writeable, possibly even from the code’s perspective…